### ISSUE TITLE
Use Object.assign instead of custom extend class

### ISSUE THREAD

**Issue by andrewtobin** [CodeSeven/toastr#463]
Just looking at the v3 code and I see that there is a custom object merge, when I think that Object.assign could be utilised instead?

https://github.com/CodeSeven/toastr/blob/v3/src/toastr.js#L660
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

**Comment by TimFerrell** [comment]
Yeah, good point. Want to implement it? :)

**Comment by andrewtobin** [comment]
Absolutely, more than happy to help.  I'll probably comment on the v3 issue, but if you guys have a list of issues/objectives to get from alpha to rc or further, I'm willing to invest some time :)

**Comment by emileber** [comment]
Won't this drop IE11 support?

http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign

**Comment by andrewtobin** [comment]
That's a fair call - instead of putting back the original code though do you think using the Mozilla polyfill would work?  Just thinking we should continue letting it use the native operation if it exists.

**Comment by emileber** [comment]
Yeah, I guess a polyfill would work.

**Comment by TimFerrell** [comment]
If we're using Babel, then babel should generate the polyfill. If it is not currently doing that, then we need to take a look at our configs.

**Comment by emileber** [comment]
I don't see any task that uses Babel, is Toastr using it?

**Comment by andrewtobin** [comment]
Babel is in the gulp build task https://github.com/CodeSeven/toastr/blob/v3/gulpfile.js#L31

I can't see anything in the final code that is a polyfill for it though, so I think Tim is right.

**Comment by TimFerrell** [comment]
I've come across this before - you need to tell Babel which polyfills to
include and where.

I'll try to get a change out today if possible. Thanks all for your
discovery and efforts.

Tim Ferrell

**Comment by emileber** [comment]
@andrewtobin thanks for the info, I'm new to this repo and didn't knew about `v3` yet.

**Comment by emileber** [comment]
@andrewtobin haaa didn't notice the `v3` branch and I was checking on master.
